### PR TITLE
Simplify unicast event routing

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds `unicast()` methods for `EventRouting` to simplify the cases when an event target is only one entity.

Previously, it was required to wrap the ID of the target entity into `EventRoute.withId()` utility call.

Only some of the `withId()` calls were replaced with simplified calls to make this PR merge sooner. 
We'll migrate to simpler API where appropriate when corresponding framework areas or tests are updated.

### CODE REVIEW

File: server/src/main/java/io/spine/server/route/EventFnRoute.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *         the type of the event message
+ */
+final class EventFnRoute<I, E extends EventMessage> implements EventRoute<I, E> {
+
+    private static final long serialVersionUID = 0L;
+    private final BiFunction<E, EventContext, I> fn;
+
+    EventFnRoute(Function<E, I> fn) {
+        checkNotNull(fn);
+        this.fn = (e, ctx) -> fn.apply(e);
```
Comment (yuri-sergiichuk) [inline server/src/main/java/io/spine/server/route/EventFnRoute.java:48]: Maybe use `this((e, ctx) -> fn.apply(e));` instead?

**Review by yuri-sergiichuk — APPROVED** [review]
@alexander-yevsyukov LGTM.

File: server/src/main/java/io/spine/server/route/EventFnRoute.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import io.spine.core.EventContext;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An event route with one target entity which ID is obtained as a function
```
Comment (dmitrykuzmin) [inline server/src/main/java/io/spine/server/route/EventFnRoute.java:33]: I think, the grammatically correct form would be `ID of which is obtained...`.

**Review by armiol — APPROVED** [review]

**Comment by alexander-yevsyukov** [comment]
51fecc6 also addresses #1316.

**Pull request merged**